lib/logstash/filters/elasticsearch.rb

```diff
@@ -38,9 +73,13 @@ class LogStash::Filters::Elasticsearch < LogStash::Filters::Base
   # Comma-delimited list of index names to search; use `_all` or empty string to perform the operation on all indices
   config :index, :validate => :string, :default => ""
 
-  # Elasticsearch query string. Read the Elasticsearch query string documentation
+  # Elasticsearch query string. Read the Elasticsearch query string documentation. Deprecated.
   # for more info at: https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-query-string-query.html#query-string-syntax
-  config :query, :validate => :string
+  config :query, :validate => :string, :deprecated => true
+
+  # File path to elasticsearch query in DSL format. Read the Elasticsearch query documentation
+  # for more info at: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html
+  config :query_template, :validate => :string
```

Comment:
Add a note in the documentation to tell people that it support the fieldref syntax.